### PR TITLE
ENH: support any/all for pyarrow numeric and duration dtypes

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Cuts the runtime for `pytest pandas/tests/extension/test_arrow.py::TestBaseBooleanReduce::test_reduce_series` from 39.8s to 2.3s.

### CODE REVIEW

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -971,6 +974,27 @@ def _reduce(self, name: str, *, skipna: bool = True, **kwargs):
...
+        ):
+            # pyarrow only supports any/all for boolean dtype, we allow
+            #  for other dtypes, matching our non-pyarrow behavior
+
+            zero: int | Timedelta
+            if pa.types.is_duration(pa_type):
+                zero = Timedelta(0)
+            else:
+                zero = 0
+            result = (self != zero)._reduce(name, skipna=skipna, **kwargs)
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:992]: Could we use `data = pc.not_equal(self._data, zero)` and continue with the logic below?

File: pandas/core/arrays/arrow/array.py [hunk 2]
```diff
@@ -1008,6 +1011,29 @@ def _reduce(self, name: str, *, skipna: bool = True, **kwargs):
...
+            # pyarrow only supports any/all for boolean dtype, we allow
+            #  for other dtypes, matching our non-pyarrow behavior
+
+            zero: int | Timedelta
+            if pa.types.is_duration(pa_type):
+                zero = Timedelta(0)
+            else:
+                zero = 0
+
+            not_eq = pc.not_equal(self._data, zero)
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:1030]: Ah sorry, I had meant 

```
if name in ...:
    ...
    data = pc.not_equal(self._data, zero)
else:
    data = self._data
....
try:
     result = pyarrow_method(data...)
```

**Comment by jbrockmendel** [comment]
updated per requests + green

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**